service/controller/resource/prometheusremotewrite/create.go

```diff
@@ -30,7 +30,7 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 
 		for _, current := range prometheusList.Items {
 
-			desired, ok := ensurePrometheusRemoteWrite(*remoteWrite, *current)
+			desired, ok := ensurePrometheusRemoteWrite(r, *remoteWrite, *current)
```

Comment:
I would recommend to make this function a method of `Resource` instead of passing `r` as an argument.

So this becomes:

```suggestion
			desired, ok := r.ensurePrometheusRemoteWrite(*remoteWrite, *current)
```